scripts/ice-road-pipeline.py

```diff
@@ -43,186 +60,211 @@
+    # ---- docopt parse and in_dir validation ----
+    try:
+        args = docopt(__doc__)
+    except SystemExit:
+        # If user requested --help, allow docopt to print full help text normally
+        if "--help" in sys.argv or "-h" in sys.argv:
+            sys.exit(0)
 
-    imu_data = args.get('-i')
-    if imu_data:
-        imu_data = abspath(imu_data)
+        print("\nERROR: Missing input directory <in_dir>.\n")
+        print("Example:\n  ice-road-pipeline.py /tmp/data --shp=road.shp --dem=ref.tif\n")
+        sys.exit(1)
 
-    cal_las = args.get('-c')
-    if cal_las:
-        cal_las = abspath(cal_las)
+    # ---- Enforce user DEM(-e) and alignment shapefile(-e) ----
+    required = ["--shp", "--dem"]
+    friendly = {
+        "--shp": "alignment shapefile (--shp=<file>)",
+        "--dem": "reference DEM (--dem=<file>)",
+    }
 
-    known_rfl = args.get('-k')
-    if known_rfl:
-        known_rfl = float(known_rfl)
+    missing = [r for r in required if not args.get(r)]
+    if missing:
+        print(f"\nERROR: Missing required argument: {friendly[missing[0]]}\n")
+        sys.exit(1)
 
-    h2o = args.get('-h')
-    if h2o:
-        h2o = float(h2o)
+    # ---- schema validation ----
+    schema = Schema({
+        "<in_dir>": And(os.path.exists, os.path.isdir,
+                        error="<in_dir> must be an existing directory"),
+        "--shp": And(os.path.exists, lambda p: p.lower().endswith('.shp'),
+                     error="--shp must be a .shp file"),
+        "--dem": And(os.path.exists, lambda p: p.lower().endswith(('.tif', '.tiff')),
+                     error="--dem must be a .tif/.tiff DEM"),
 
-    aod = args.get('-o')
-    if aod:
-        aod = float(aod)
+        "--buffer": Or(None, Use(float, error="--buffer must be numeric")),
+        "--smrf-scalar": Or(None, Use(float, error="--smrf-scalar must be numeric")),
+        "--smrf-slope": Or(None, Use(float, error="--smrf-slope must be numeric")),
+        "--smrf-threshold": Or(None, Use(float, error="--smrf-threshold must be numeric")),
+        "--smrf-window": Or(None, Use(float, error="--smrf-window must be numeric")),
 
-    smrf_overrides = {}
-    for flag, key in SMRF_OPTIONS:
-        value = args.get(flag)
-        if value is not None:
-            smrf_overrides[key] = float(value)
+        "--debug": bool,
+        "--geoid": bool,
+        "--skip-filter": bool,
+        "--force": bool,
 
-    use_dem_filter = not args['-D']
+        "--asp-dir": Or(None, str),
+        "--shp-rfl": Or(None, str),
+        "--imu": Or(None, str),
+        "--cal-las": Or(None, str),
+        "--known-rfl": Or(None, Use(float)),
+        "--h2o": Or(None, Use(float)),
+        "--aod": Or(None, Use(float)),
+    })
 
-    in_dir = args.get('<in_dir>')
-    # convert to abspath
-    in_dir = abspath(in_dir)
-    
+    try:
+        args = schema.validate(args)
+    except SchemaError as e:
+        print(f"\nSCHEMA ERROR: {e}\n")
```

Comment:
Very cool use of Schema! ❤️ 

Maybe make this a more user known error message and say "Argument error". Not sure that a standard user would know what to do with a "Schema error".